dialect/sql/schema/mysql.go

```diff
@@ -449,6 +435,39 @@ func (d *MySQL) tableSchema() sql.Querier {
 	return sql.Raw("(SELECT DATABASE())")
 }
 
+// columnData returns column data: parts, size and signedness
+func columnData(typ string) ([]string, int64, bool, error) {
+	parts := strings.FieldsFunc(typ, func(r rune) bool {
+		return r == '(' || r == ')' || r == ' ' || r == ','
+	})
+	switch parts[0] {
+	case "int", "smallint", "bigint", "tinyint":
+		switch {
+		case len(parts) == 2 && parts[1] == "unsigned": // int unsigned
+			return parts, 0, true, nil
+		case len(parts) == 2: // int(10)
+			size, err := strconv.ParseInt(parts[1], 10, 0)
+			if err != nil {
+				return parts, 0, false, fmt.Errorf("converting %s size to int: %w", parts[0], err)
+			}
+			return parts, size, false, nil
+		case len(parts) == 3: // int(10) unsigned
+			size, err := strconv.ParseInt(parts[1], 10, 0)
+			if err != nil {
+				return parts, 0, true, fmt.Errorf("converting %s size to int: %w", parts[0], err)
+			}
+			return parts, size, true, nil
+		}
```

Comment:
Let's avoid the duplication here, and change it to something like:
```go
var (
  size int
  unsigned bool
)
switch {
	case len(parts) == 2 && parts[1] == "unsigned": // int unsigned
		unsigned = true
	case len(parts) == 3: // int(10) unsigned
		unsigned = true
		fallthrough 
	case len(parts) == 2: // int(10)
		size, err = strconv.ParseInt(parts[1], 10, 0)
		if err != nil {
			return parts, 0, false, fmt.Errorf("converting %s size to int: %v", parts[0], err)
		}
}
```

(`size` and `unsigned` can be used by the second switch as well).